bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/utils/JsonUtils.java

```diff
@@ -258,6 +277,114 @@ public static void serialize(Object json, Writer writer) throws BError {
         }
     }
 
+    public static Object convertToJson(Object value, List<TypeValuePair> unresolvedValues) {
+        Type jsonType = PredefinedTypes.TYPE_JSON;
+
+        if (value == null) {
+            return null;
+        }
+
+        Type sourceType = TypeChecker.getType(value);
+
+        if (sourceType.getTag() <= TypeTags.BOOLEAN_TAG && TypeChecker.checkIsType(value, jsonType)) {
+            return value;
+        }
+
+        TypeValuePair typeValuePair = new TypeValuePair(value, jsonType);
+
+        if (unresolvedValues.contains(typeValuePair)) {
+            throw createCyclicValueReferenceError(value);
+        }
+
+        unresolvedValues.add(typeValuePair);
+
+        Object newValue;
+        switch (sourceType.getTag()) {
+            case TypeTags.XML_TAG:
+            case TypeTags.XML_ELEMENT_TAG:
+            case TypeTags.XML_COMMENT_TAG:
+            case TypeTags.XML_PI_TAG:
+            case TypeTags.XML_TEXT_TAG:
+                newValue = StringUtils.fromString(StringUtils.getStringValue(value, null));
+                break;
+            case TypeTags.TUPLE_TAG:
+            case TypeTags.ARRAY_TAG:
+                newValue = convertArrayToJson((BArray) value, unresolvedValues);
+                break;
+            case TypeTags.TABLE_TAG:
+                BTable bTable = (BTable) value;
+                Type constrainedType = ((TableType) bTable.getType()).getConstrainedType();
+                if (constrainedType.getTag() == TypeTags.MAP_TAG) {
+                    newValue = convertMapConstrainedTableToJson((BTable) value, unresolvedValues);
+                } else {
+                    try {
+                        newValue = io.ballerina.runtime.internal.JsonUtils.toJSON(bTable);
```

Comment:
Shall we import `io.ballerina.runtime.internal.JsonUtils` and call as `JsonUtils.toJSON` here for readability? There are more places like this below.